launchable/utils/session.py

```diff
@@ -1,102 +1,103 @@
 import os
-import sys
 from pathlib import Path
 from typing import Optional
-import datetime
-import hashlib
-from .logger import Logger
+import json
 
 SESSION_DIR_KEY = 'LAUNCHABLE_SESSION_DIR'
-DEFAULT_SESSION_DIR = '~/.config/launchable/sessions/'
 
 
 def _session_file_dir() -> Path:
-    return Path(os.environ.get(SESSION_DIR_KEY) or DEFAULT_SESSION_DIR).expanduser()
-
-
-def _session_file_path(build_name: str) -> Path:
-    return _session_file_dir() / (hashlib.sha1("{}:{}".format(build_name, _get_session_id()).encode()).hexdigest() + ".txt")
-
-
-def _get_session_id() -> str:
-    # CircleCI changes unix session id each steps, so set non change variable
-    # https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables
-    if os.environ.get("CIRCLECI") is not None:
-        id = os.environ.get("CIRCLE_WORKFLOW_ID")
-        if id is not None:
-            return id
-        Logger().warning("CIRCLECI environment variable is set but not CIRCLE_WORKFLOW_ID")
-
-    # Jenkins pipeline also launches every process as a new session, so better to scope this to a particular build.
-    # For freestyle job types that do not do this, session IDs do not distinguish different builds, so that is a problem, too.
-    #
-    # See: https://github.com/jenkinsci/lib-durable-task/blob/6e020747205cb5aca5af757bc0d2a302c42bdb79/src/cmd/bash/durable_task_monitor.go#L41
-    # See: https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
-    if os.environ.get("JENKINS_URL") is not None:
-        id = os.environ.get("BUILD_URL")
-        if id is not None:
-            return id
-        Logger().warning("JENKINS_URL environment variable is set but not BUILD_URL")
-
-    if sys.platform == "win32":
-        import wmi  # type: ignore
-        c = wmi.WMI()
-        wql = "Associators of {{Win32_Process='{}'}} Where Resultclass = Win32_LogonSession Assocclass = Win32_SessionProcess".format(
-            os.getpid())
-        res = c.query(wql)
-        return str(res[0].LogonId)
-    else:
-        return str(os.getsid(os.getpid()))
+    return Path(os.environ.get(SESSION_DIR_KEY) or os.getcwd()).expanduser()
+
+
+def _session_file_path() -> Path:
+    return _session_file_dir() / ".launchable"
+
+
+def read_build() -> Optional[str]:
+    f = _session_file_path()
+    try:
+        if not f.exists():
+            return None
+
+        with open(str(_session_file_path())) as session_file:
+            session = json.load(session_file)
+            return session["build"]
```

Comment:
This line will raise `KeyError` if the session doesn't contain build, whiich is probably not what we want. I think `session["build"]` is what we want:
```
>>> x=json.loads('{"a":3}')
>>> x["b"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'b'
>>> x.get("b")
>>> 
```